### PR TITLE
PS-6052: gcc 9.2.1 reports -Werror=stringop-overflow in os_file_get_parent_dir (8.0)

### PR DESCRIPTION
**Pull request by inikep**

### CODE REVIEW

File: storage/innobase/os/os0file.cc [hunk 1]
```diff
@@ -1990,6 +1990,10 @@ static char *os_file_get_parent_dir(const char *path) {
     return (NULL);
   }
 
+  if (last_slash - path < 0) {
```
Comment (satya-bodapati) [inline storage/innobase/os/os0file.cc:1994]: what does this code mean/ does? both are char pointers

Comment (satya-bodapati) [reply storage/innobase/os/os0file.cc:1994]: Please add comment (like it was done above) :)

**Review by satya-bodapati — CHANGES_REQUESTED** [review]

**Comment by inikep** [comment]
Tested at https://ps80.cd.percona.com/job/percona-server-8.0-param/399/

File: storage/innobase/os/os0file.cc (hunk 1 above)
Comment (inikep) [reply storage/innobase/os/os0file.cc:1994]: done

**Review by percona-ysorokin — APPROVED** [review]
LGTM

**Pull request merged**